eng/Versions.props

```diff
@@ -123,13 +123,17 @@
     <MicrosoftWin32RegistryVersion>5.0.0</MicrosoftWin32RegistryVersion>
     <StyleCopAnalyzersVersion>1.2.0-beta.507</StyleCopAnalyzersVersion>
     <SystemBuffersVersion>4.5.1</SystemBuffersVersion>
-    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
+    <SystemCollectionsImmutableVersion>9.0.0-alpha.1.24064.3</SystemCollectionsImmutableVersion>
```

Comment:
What is the intent here?  Is this supposed to be the live version?  How will this get updated?

Reply:
Currently it is pinned; we manually try to keep sync it with sdk and msbuilid, see https://github.com/dotnet/runtime/pull/97021#issuecomment-1904515180. This PR is separating 'pinned version' (used in two places) vs. 'floating/auto-updating version' (used elsewhere), so we can consume the newer version where pinning is not needed. Ideally, we should be able to use `<ProjectReference Include="$(LibrariesProjectRoot)System.Reflection.Metadata\src\System.Reflection.Metadata.csproj" />` across the monorepo, but because the build is order-dependent, we cannot build R2R projects with live version; with the current order at least, it's not possible AFAIK, we need to build the libs subset before clr.

Reply:
I suspect this is going to cause a prebuilt in source-build and adding an exclusion would not be appropriate as @ViktorHofer indicated in his earlier comment.  It will get the repo source-build CI to be green but will break source-build when this change flows into installer. 

If a project to project dependency can not be declared, these dependencies must be declared in the versions.details.xml file.  Having these dependencies defined signals to source-build that the versions can be upgraded to current (n-1 or n depending on if the component has been built yet or not).  In the repo CI this should also eliminate the prebuilts from being reported.

Reply:
This version should be updated to match the version.details.xml file.

Reply:
Thank you for the pointers! It has eliminated the prebuilt versions in SB leg. 👍